benchmarks/float8/bench_linear_float8.py

```diff
@@ -112,137 +81,93 @@ def main(
-        cast_config_grad_output=CastConfig(
-            scaling_type=scaling_type_grad_output,
-            static_scale=torch.tensor([1.0], device="cuda"),
-            scaling_granularity=scaling_granularity,
-        )
-    else:
-        cast_config_grad_output=CastConfig(
-            scaling_type=scaling_type_grad_output,
-            scaling_granularity=scaling_granularity,
+    if recipe_name is None:
+        config = get_test_float8_linear_config(
+            scaling_type_input,
+            scaling_type_weight,
+            scaling_type_grad_output,
+            emulate=False,
         )
-
-    config = Float8LinearConfig(
-        cast_config_input=cast_config_input,
-        cast_config_weight=cast_config_weight,
-        cast_config_grad_output=cast_config_grad_output,
-    )
+    elif recipe_name is not None:
+        recipe_name = Float8LinearRecipeName(recipe_name)
+        config = recipe_name_to_linear_config(recipe_name)
+    print(f"config: {config}")
 
     name_to_shapes = get_name_to_shapes_iter(shape_gen_name, M, K, N)
     input_bias = False
-    if fast_accum_filter is not None:
-        use_fast_accum = [fast_accum_filter]
-    else:
-        use_fast_accum = [True, False]
     if shape_name_filter is not None:
         k = shape_name_filter
         name_to_shapes = ((k, v) for (k, v) in name_to_shapes if k == shape_name_filter)
     experiment_list: List[Experiment] = []
     dtype = torch.bfloat16
-    for idx, (fast_accum, (name, (M, K, N))) in enumerate(
-        tqdm(list(product(use_fast_accum, name_to_shapes)))
-    ):
+    for idx, (name, (M, K, N)) in enumerate(tqdm(name_to_shapes)):
         if n_limit is not None and idx >= n_limit:
             break
-        linear_ref = torch.nn.Linear(K, N, bias=input_bias).to(
-            device=device, dtype=dtype
-        )
+        m_ref, input_tensors = subgraph_name_to_subgraph_and_inputs(
+            model_type, mkn_override=(M, K, N))
 
-        linear_float8 = Float8Linear.from_float(
-            copy.deepcopy(linear_ref),
-            config=config,
-        )
-        scaling_repr = f"{linear_float8.scaling_type_repr()},{linear_float8.scaling_granularity_repr()}"
+        # linear_ref = nn.Sequential(nn.Linear(K, N, bias=input_bias)).to(
+        #     device=device, dtype=dtype
+        # )
+        # input_tensor = torch.randn(M, K, device=device, dtype=dtype, requires_grad=True)
 
-        if fast_accum:
-            linear_float8.forward_config = ScaledMMConfig(False, True, False)
-        else:
-            linear_float8.forward_config = ScaledMMConfig(False, False, False)
+        m_float8 = copy.deepcopy(m_ref)
+        convert_to_float8_training(m_float8, config=config)
 
-        input_tensor = torch.randn(M, K, device=device, dtype=dtype, requires_grad=True)
-        ref_forw_backward = lambda: linear_ref(input_tensor).sum().backward()
+        def ref_forw_backward(*input_tensors):
+            return m_ref(*input_tensors).sum().backward()
```

Comment:
The sum_backwards can have some interesting side effects because the back propping grad is broadcasted, maybe: torch.empty_like(out)